### DISCUSSION TITLE
3. 데이터베이스 - SQL 고급

### DISCUSSION THREAD

**Discussion by zini9188** [Shortudy-10th/tech-for-developer#152] [질의 응답]
- 부속질의
- 뷰
- 인덱스
- 그 외 등등

**Comment by zini9188** [comment]
### Q. 데이터베이스에서 뷰는 무엇인가요?
하나 이상의 테이블이나 다른 뷰의 데이터를 볼 수 있게 하는 가상의 테이블로 실제 데이터를 저장하지는 않지만 일반 테이블과 동일한 방법으로 사용할 수 있습니다.

**Reply by zini9188** [reply]
### Q. 뷰를 사용하는 이유는 무엇인가요?
복잡한 조인문이 생기는 쿼리를 뷰에 저장하여 간단하게 사용할 수 있으며 컬럼과 데이터만을 공개하므로 원천 테이블 정보를 알 수 없어 보안에 유리합니다.

**Comment by seongeun42** [comment]
## Q. 서브쿼리란 무엇인가요?

하나의 SQL문 안에 포함되어 있는 또 다른 SQL문을 말합니다. 필요한 값을 조회한 후 실제로 조회하고자 하는 값을 찾는데 사용됩니다.
